### PR TITLE
feat(nutanix): add nodes sections on nutanix dashboard

### PR DESCRIPTION
**Pull request by tibs245**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/nutanix`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-14695
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/manager/modules/nutanix/src/node.class.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { NODE_STATUS } from './constants';
+
+export default class Cluster {
```
Comment (darsene) [inline packages/manager/modules/nutanix/src/node.class.js:3]:
```suggestion
export default class Node {
```

File: packages/manager/modules/nutanix/src/service.js [hunk 2]
```diff
@@ -169,6 +169,16 @@ export default class NutanixService {
       .then((res) => res);
   }
 
+  getNodeWithState(serviceName) {
```
Comment (darsene) [inline packages/manager/modules/nutanix/src/service.js:172]:
```suggestion
  getNodesWithState(serviceName) {
```

File: packages/manager/modules/nutanix/src/service.js [hunk 3]
```diff
@@ -169,6 +169,16 @@ export default class NutanixService {
...
+    return this.$http
+      .get(`/nutanix/${serviceName}/nodes`, {
+        serviceType: 'aapi',
+      })
+      .then(({ data }) => {
+        data.map((node) => new Node(node));
```
Comment (darsene) [inline packages/manager/modules/nutanix/src/service.js:178]: You can create a class Cluster too

```
class Cluster {

constructor({nodes}){
   this.nodes = nodes;
}
```

File: packages/manager/modules/nutanix/src/dashboard/general-info/controller.js [hunk 4]
```diff
@@ -61,6 +63,27 @@ export default class NutanixGeneralInfoCtrl {
...
+
+  get numberNodeDeployed() {
+    if (this.loadingNodeStatus) return '-';
+    return this.nodes.filter((node) => node.isDeployed).length;
+  }
+
+  get numberNodeToDeploy() {
+    if (this.loadingNodeStatus) return '-';
+    return this.nodes.filter((node) => !node.isDeployed).length;
+  }
```
Comment (darsene) [inline packages/manager/modules/nutanix/src/dashboard/general-info/controller.js:85]: To move to the Cluster class

File: packages/manager/modules/nutanix/src/dashboard/general-info/template.html [hunk 5]
```diff
@@ -128,6 +128,53 @@
...
             </oui-tile-definition>
 
+            <!-- number of nodes -->
+            <oui-tile-definition
+                term="{{:: 'nutanix_dashboard_cluster_number_of_nodes' | translate }}"
+            >
+                <oui-tile-description class="align-items-start">
+                    <div class="text-break">
+                        <span class="font-weight-bold text-break"
+                            >{{ $ctrl.nodes.length }}</span
```
Comment (darsene) [inline packages/manager/modules/nutanix/src/dashboard/general-info/template.html:138]:
```suggestion
                        <span class="font-weight-bold text-break"
                        data-ng-bind="$ctrl.nodes.length"
                            ></span
```

File: packages/manager/modules/nutanix/src/dashboard/general-info/template.html [hunk 6]
```diff
@@ -128,6 +128,53 @@
...
+                        >
+                        {{ "nutanix_dashboard_cluster_total_suscribed" |
+                        translate }}
```
Comment (darsene) [inline packages/manager/modules/nutanix/src/dashboard/general-info/template.html:141]:
```suggestion
                       <span data-translate="nutanix_dashboard_cluster_total_suscribed"></span>
```

File: packages/manager/modules/nutanix/src/dashboard/general-info/template.html [hunk 7]
```diff
@@ -128,6 +128,53 @@
...
+                    <div>
+                        <span
+                            class="oui-badge oui-badge_success text-break ml-n1"
+                            data-translate="nutanix_dashboard_cluster_node_deployed"
+                            data-translate-values="{ numberOfNodes: $ctrl.numberNodeDeployed, suffixPlurial: ($ctrl.numberNodeDeployed > 1 ? 's' : '') }"
+                        ></span>
+                    </div>
+                    <div>
+                        <a class="oui-link_icon" href="#">
+                            Gérer mes noeuds
```
Comment (darsene) [inline packages/manager/modules/nutanix/src/dashboard/general-info/template.html:159]: use translation. And there is no href link define here, is it normal ?

**Review by darsene — CHANGES_REQUESTED** [review]

File: packages/manager/modules/nutanix/src/dashboard/general-info/translations/Messages_fr_FR.json [hunk 8]
```diff
@@ -8,6 +8,8 @@
   "nutanix_dashboard_cluster_admin_interface_info": "Veuillez vous connecter via Prism Central pour gérer votre cluster Nutanix",
   "nutanix_dashboard_cluster_access_admin_interface": "Accéder à Prism Central",
   "nutanix_dashboard_cluster_type_of_pack": "Type de pack",
+  "nutanix_dashboard_cluster_number_of_nodes": "Nombre de noeuds",
+  "nutanix_dashboard_cluster_buy_node": "Commande un noeud",
```
Comment (ThibaudCrespin) [inline packages/manager/modules/nutanix/src/dashboard/general-info/translations/Messages_fr_FR.json:12]:
```suggestion
  "nutanix_dashboard_cluster_buy_node": "Commander un noeud",
```

File: packages/manager/modules/nutanix/src/node.class.js (hunk 1 above)
Comment (tibs245) [reply packages/manager/modules/nutanix/src/node.class.js]: Seen together. Cluster already exists but is in conflict with API data

**Review by ThibaudCrespin — DISMISSED** [review]

File: packages/manager/modules/nutanix/src/dashboard/general-info/translations/Messages_fr_FR.json [hunk 9]
```diff
@@ -20,5 +23,8 @@
   "nutanix_dashboard_cluster_redeploy_modal_header": "Redéploiement du cluster",
   "nutanix_dashboard_cluster_redeploy_modal_description": "Il est nécessaire de désinstaller votre licence Nutanix de votre cluster dans Prism Central avant le redéploiement de votre cluster.",
   "nutanix_dashboard_cluster_redeploy_modal_continue": "Continuer",
-  "nutanix_dashboard_cluster_redeploy_modal_cancel": "Annuler"
+  "nutanix_dashboard_cluster_redeploy_modal_cancel": "Annuler",
+  "nutanix_dashboard_cluster_total_suscribed": "noeuds dans la souscription initale",
+  "nutanix_dashboard_cluster_node_to_deploy": "{{numberOfNodes}} noeud{{suffixPlurial}} à configurer",
+  "nutanix_dashboard_cluster_node_deployed": "{{numberOfNodes}} noeud{{suffixPlurial}} installé{{suffixPlurial}}"
```
Comment (darsene) [inline packages/manager/modules/nutanix/src/dashboard/general-info/translations/Messages_fr_FR.json:29]: It doesn't work for all the languages, `s` is not always the suffix for plural. By the way, the variable should be `suffixPural`

File: packages/manager/modules/nutanix/src/dashboard/general-info/template.html [hunk 10]
```diff
@@ -128,6 +128,58 @@
...
+                            class="oui-badge oui-badge_warning text-break ml-n1"
+                            data-translate="nutanix_dashboard_cluster_node_to_deploy"
+                            data-translate-values="{ numberOfNodes: $ctrl.numberNodesToDeploy, suffixPlurial: ($ctrl.numberNodesToDeploy > 1 ? 's' : '') }"
+                        ></span>
+                    </div>
+                    <div>
+                        <span
+                            class="oui-badge oui-badge_success text-break ml-n1"
+                            data-translate="nutanix_dashboard_cluster_node_deployed"
+                            data-translate-values="{ numberOfNodes: $ctrl.numberNodesDeployed, suffixPlurial: ($ctrl.numberNodesDeployed > 1 ? 's' : '') }"
```
Comment (darsene) [inline packages/manager/modules/nutanix/src/dashboard/general-info/template.html:156]:
```suggestion
                            <strong
                    data-ng-pluralize
                    data-count="$ctrl.numberNodesDeployed"
                    data-when="{
                        'one': ('nutanix_dashboard_cluster_one_node_to_deploy'  | translate),
                        'other': ('nutanix_dashboard_cluster_several_nodes_to_deploy'  | translate),
                    }"
                >
                </strong>"
```

**Review by darsene — CHANGES_REQUESTED** [review]

**Review by darsene — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Review by ThibaudCrespin — APPROVED** [review]

**Pull request merged**